app/controllers/users/passwords_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Users::PasswordsController < Devise::PasswordsController
+  include ApplicationHelper
+  include PhoneNumberHelper
+  include SmsBodyHelper
+
+  def create
+    email = params[resource_name][:email]
+    phone_number = params[resource_name][:phone_number]
+    reset_token = nil
+
+    if email.blank? && phone_number.blank?
+      resource.errors.add(:base, "Please enter at least one field.")
+    end
+
+    if !email.blank? && !User.find_by(email: email)
+      resource.errors.add(:base, "Email not found")
+    end
+
+    validation = valid_phone_number(phone_number)
+    if validation[0]
+      User.find_by(phone_number: phone_number) ? "" : resource.errors.add(:base, "Phone number not found")
+    else
+      resource.errors.add(:phone_number, validation[1])
+    end
+
+    # re-render and display any errors
+    if resource.errors.any?
+      respond_with(resource)
+      return
+    end
+
+    # otherwise, send reset email and sms
+    @resource = email.blank? ? User.find_by(phone_number: phone_number) : User.find_by(email: email)
```

Comment:
Seems like you are already querying `User.find_by(phone_number: phone_number)` and `User.find_by(email: email)` above in the file to add error if the queries doesn't find anything. Maybe put these in a variable so you only call them once and reuse them if the query does find something?